### PR TITLE
Don't define label encoder for binary logistic.

### PR DESCRIPTION
**Pull request by trivialfis**
During testing #5277 , I found the generated model is extremely huge and slow.  The result is binary logistic doesn't require label to be strictly in {0, 1}.  Anything fits into sigmoid is accepted.  So running label encoding will simply return the labels in worse case.

### CODE REVIEW

File: python-package/xgboost/sklearn.py [hunk 1]
```diff
@@ -749,7 +749,10 @@ def fit(self, X, y, sample_weight=None, base_margin=None,
         # pylint: disable = attribute-defined-outside-init,arguments-differ
 
         evals_result = {}
-        self.classes_ = np.unique(y)
+        if self.objective != 'binary:logistic':
+            self.classes_ = np.array([0.0, 1.0])
```
Comment (hcho3) [inline python-package/xgboost/sklearn.py]: ??? Why 0 and 1 here?

Comment (trivialfis) [reply python-package/xgboost/sklearn.py]: @hcho3  Sorry, still writing tests.  Just raised the PR early.

**Comment by trivialfis** [comment]
@hcho3 It's a little bit tricky.  If we don't run `np.unique`, we don't know how many classes are there, hence we can't decide the objective.  But if we run it on a binary classification model, `np.unique` is not accurate.

**Comment by hcho3** [comment]
> But if we run it on a binary classification model, np.unique is not accurate

Why? You’d still get two classes in this case?

**Comment by trivialfis** [comment]
@hcho3 The issue is, `y` is not necessarily integers or categorical.  It can be floating point numbers.

**Comment by trivialfis** [comment]
@hcho3 We can throw a warning for non-discrete types.  But since the input varies as there are numpy, pandas, and possibly others may also works with Python duck typing.  Checking types is not easy either.  Do you have any suggestion?
